src/highlight.js

```diff
@@ -3,34 +3,19 @@ Syntax highlighting with language autodetection.
 https://highlightjs.org/
 */
 
-(function(factory) {
-
-  // Find the global object for export to both the browser and web workers.
-  var globalObject = typeof window === 'object' && window ||
-                     typeof self === 'object' && self;
-
-  // Setup highlight.js for different environments. First is Node.js or
-  // CommonJS.
-  // `nodeType` is checked to ensure that `exports` is not a HTML element.
-  if(typeof exports !== 'undefined' && !exports.nodeType) {
-    factory(exports);
-  } else if(globalObject) {
-    // Export hljs globally even when using AMD for cases when this script
-    // is loaded with others that may still expect a global hljs.
-    globalObject.hljs = factory({});
-
-    // Finally register the global hljs with AMD.
-    if(typeof define === 'function' && define.amd) {
-      define([], function() {
-        return globalObject.hljs;
-      });
-    }
-  }
+import deepFreeze from './vendor/deep_freeze';
```

Comment:
What's the reason to put it into `vendor` (instead of `utils`)?

Reply:
It's pretty much borrowed code from elsewhere, hence vendoring it.

Reply:
I don't insist, although for me a comment pointing to the original repo would be enough.

Are you done with the PR or you're going to add some more changes?

Reply:
It's a Rails habit, but I like it.  Doesn't hurt anything to organize it nicely.

> Are you done with the PR or you're going to add some more changes?

Done.  I try to not assign things or label them WIP if they aren't finished.

Reply:
This finally lays the groundwork for a parse tree, but that's another PR.